File: samples/typescript/src/streaming.ts
// ストリーミング（SSE）サンプル。
// 実行: npm run streaming
import OpenAI from "openai";

const client = new OpenAI({
  apiKey: process.env.LYKURO_API_KEY,
  baseURL: process.env.LYKURO_BASE_URL ?? "https://api.lykuro.ai/deepseek/v1",
});

const stream = await client.chat.completions.create({
  model: "deepseek-chat",
  messages: [{ role: "user", content: "春をテーマに俳句を3つ作ってください。" }],
  stream: true,
});

for await (const chunk of stream) {
  const delta = chunk.choices[0]?.delta?.content;
  if (delta) process.stdout.write(delta);
}
process.stdout.write("\n");

File: samples/python/streaming.py
"""ストリーミング（SSE）サンプル。"""
import os

from dotenv import load_dotenv
from openai import OpenAI

load_dotenv()

client = OpenAI(
    api_key=os.environ["LYKURO_API_KEY"],
    base_url=os.environ.get("LYKURO_BASE_URL", "https://api.lykuro.ai/deepseek/v1"),
)


def main() -> None:
    stream = client.chat.completions.create(
        model="deepseek-chat",
        messages=[{"role": "user", "content": "春をテーマに俳句を3つ作ってください。"}],
        stream=True,
    )
    for chunk in stream:
        delta = chunk.choices[0].delta.content
        if delta:
            print(delta, end="", flush=True)
    print()


if __name__ == "__main__":
    main()
